manifests/init.pp

```diff
@@ -1,16 +1,33 @@
 class clamps (
-  $num_dynamic_files = 5,
-  $num_static_files = 20,
+  $num_dynamic_files     = 5,
+  $num_static_files      = 20,
+  $num_static_files_step = 5,
 ) {
 
   file { [
       "/home/${id}/clamps_files/",
       "/home/${id}/clamps_files/static/",
       "/home/${id}/clamps_files/dynamic/",
+      "/home/${id}/.facter",
+      "/home/${id}/.facter/facts.d"
     ]:
     ensure => directory,
   }
 
+  $static_files_fact_path = "/home/${id}/.facter/facts.d/static_files.txt"
+
+  exec { 'create static_files fact' :
+    command => "/bin/echo static_files=50 > ${static_files_fact_path}",
+    creates => $static_files_fact_path,
+  }
+
+  $num_static_files_to_make_now = min( pick($::static_files, 0) + $num_static_files_step, $num_static_files)
+
+  file { $static_files_fact_path :
+    ensure  => present,
+    content => "static_files=${num_static_files_to_make_now}",
+  }
+
```

Comment:
My puppet-fu is weak enough to not understand how the `Exec` and the `File` resource interact here on this same file. I'm presuming it's "if the file doesn't exist, create it with a setting of 50", and "if the file does exist, ensure that the content has a computed value based on our stepwise increase"?